src/Makefile.am

```diff
@@ -363,6 +374,27 @@ libbitcoin_wallet_tool_a_SOURCES = \
   wallet/wallettool.cpp \
   $(BITCOIN_CORE_H)
 
+# rustc only reqires the main .rs file as input, but rustc must also be invoked
+# if any of the other files change. List the main file first, then targets can
+# depend on $(LIBBITCOIN_RUSTY_SRCS), but only build the first source file ($<).
+LIBBITCOIN_RUSTY_SRCS= \
+  rusty/src/lib.rs \
+  rusty/src/bridge.rs \
+  rusty/src/dns_headers.rs
+
+LIBBITCOIN_RUSTY_TEST_SRCS= \
+  rusty/src/test_bridge.rs
+
+# We force overflow checks on so that any integer overflows result in a panic
+# instead of, eg, attempting to alloc 4GB of memory (though we will catch such
+# an alloc and refuse it, generating a panic instead).
+# Also force panics to unwind (instead of calling abort()) as we want to catch
+# panics in the Rust threads and simply close the thread, instead of bringing
+# down the rest of Bitcoin Core.
+$(LIBBITCOIN_RUSTY): $(LIBBITCOIN_RUSTY_SRCS)
+	$(RUSTC) --crate-name rusty $< --target=$(RUST_TARGET) --crate-type=staticlib --emit=link -g -O -C overflow-checks=on -C panic=unwind --out-dir $(@D)
```

Comment:
Debugging with `gdb` doesn't seem to work with `-O`. Perhaps drop it when `--enable-debug`?